monitoring/trace.go

```diff
@@ -18,9 +18,23 @@ import (
 	"context"
 )
 
+// startSpanFunc is the signature of a function which can start tracing spans.
+type startSpanFunc func(context.Context, string) (context.Context, func())
+
+var startSpan startSpanFunc = func(ctx context.Context, name string) (context.Context, func()) { return ctx, func() {} }
```

Comment:
You might move the remark about no-op below up here to indicate why 'startSpan' does so little.
Maybe rename it as noopStartSpan.

Reply:
I think having the comment where it is on the exported func is good so that the behaviour is discoverable via godoc, but naming the default impl as a self-describing this is a really good suggestion - thanks!

Done.